### PR TITLE
CMake: Use LTO on all compilers where supported

### PR DESCRIPTION
**Pull request by glebm**
Bumps minimum required CMake version to 3.10 so that we actually get LTO on compilers other than the Intel Linux compiler.

3.9 would be sufficient as well.

https://cmake.org/cmake/help/v3.9/policy/CMP0069.html

### CODE REVIEW

**Comment by AJenbo** [comment]
We don't target any systems (including Android and other WIP ports) that don't have CMake 3.10 available so if bumping CMake requirements to that helps clean up things I'm happy for that to happen.

**Comment by glebm** [comment]
OK! I thought we needed to support 3.7 because of https://github.com/diasurgical/devilutionX/pull/50

**Comment by AJenbo** [comment]
Mostly it was just trivial to go all the way to 3.7 since it was what we where using previously, and low version requirements would potentially have helped future ports. But nothing has come up (AFAIK) that made 3.7-3.9 relevant. At best it made compiling on Debian 9 easier, but then there libsodium is to old any way.

File: .editorconfig [hunk 1]
```diff
@@ -16,3 +16,10 @@ end_of_line = lf
 
 [.clang-format]
 end_of_line = lf
+
+[.gitignore]
+end_of_line = lf
+
+[CMakeLists.txt]
+indent_style = space
+indent_size = 2
```
Comment (AJenbo) [inline .editorconfig]: Nice :)

File: CMakeLists.txt [hunk 2]
```diff
@@ -28,7 +28,16 @@ if(BINARY_RELEASE)
...
-  set(CMAKE_INTERPROCEDURAL_OPTIMIZATION ON)
+
+  # Use LTO on all compilers where it is supported.
+  include(CheckIPOSupported)
+  check_ipo_supported(RESULT result)
+  if(result)
+    set(CMAKE_INTERPROCEDURAL_OPTIMIZATION ON)
+    message(STATUS "LTO enabled")
+  else()
+    message(WARNING "LTO not supported by this compiler and/or CMake version")
```
Comment (AJenbo) [inline CMakeLists.txt]: Cant decide if this is needed or not.

Comment (glebm) [reply CMakeLists.txt]: This should be rare nowadays but I guess it's useful to get this warning if you're building a release and somehow have a very old compiler

Comment (AJenbo) [reply CMakeLists.txt]: Yeah guess it works as a "things could be better" notice. But then do we need the "LTO enabled" message :D

Comment (glebm) [reply CMakeLists.txt]: Probably not, feel free to remove it :)

**Pull request merged**